python/kotekan/config.py

```diff
@@ -0,0 +1,172 @@
+                sys.stderr.write("yamllint not installed, skipping pre-validation\n")
+            else:
+                sys.stderr.write("error with yamllint, skipping pre-validation\n")
+
+        try:
+            with open(file_name_full, "r") as stream:
+                config_yaml = yaml.safe_load(stream)
+        except IOError as err:
+            raise IOError(
+                "Error reading file " + file_name_full + ": " + str(err)
+            ) from err
+        except yaml.YAMLError as err:
+            raise yaml.YAMLError("Error parsing yaml: \n" + str(err) + "\n") from err
+
+        if dump:
+            sys.stderr.write(yaml.dump(config_yaml) + "\n")
+
+    else:
+        if jinja2 is None:
+            raise ImportError("jinja2 is required for processing '.j2' files")
+
+        # Load the template
+        env = jinja2.Environment(
+            loader=jinja2.FileSystemLoader(directory),
+            autoescape=jinja2.select_autoescape(),
+        )
+        template = env.get_template(file_name)
+
+        # Parse the optional variables (if any)
+        options_dict = {}
+        if jinja_options:
+            options_dict = json.loads(str(jinja_options))
+
+        # Convert to yaml
+        config_yaml_raw = template.render(options_dict)
+
+        # Dump the rendered yaml file if requested
+        if dump:
+            sys.stderr.write(config_yaml_raw + "\n")
+
+        # TODO Should we also lint the output of the template?
+        config_yaml = yaml.safe_load(config_yaml_raw)
+
+    if return_dict:
+        return config_yaml
+
+    return json.dumps(config_yaml)
+
+
+if __name__ == "__main__":
+    """
+    This script converts YAML or Jinja files into JSON for use by kotekan
+
+    All files without a ".j2" extension are treated as YMAL, files with a ".j2"
+    extension are processed by the Jinja template render.
+
+    Extra variables can be passed in with the -e flag, in json/dict format:
+     -e '{"my_val": 20, "my_val2": 30, "my_array": [0,2,1]}'
+
+    The JSON is returned in STDOUT, and error messages are returned in STDERR
+    """
+
+    parser = argparse.ArgumentParser(
+        description="Convert YAML or Jinja files into JSON"
+    )
+    parser.add_argument("name", help="Config file name", type=str)
+    parser.add_argument(
+        "-d", "--dump", help="Dump the yaml, useful with .j2 files", action="store_true"
+    )
+    parser.add_argument(
+        "-e", "--variables", help="Add extra jinja variables, JSON format", type=str
+    )
+    parser.add_argument(
+        "-s", "--stderr", help="Print exceptions to stderr", action="store_true"
+    )
+    args = parser.parse_args()
+
+    options = args.variables
+
+    try:
+        print(load_config_file(args.name, dump=args.dump, jinja_options=options))
+    except (IOError, yaml.YAMLError, ImportError, jinja2.TemplateNotFound) as err:
+        if args.stderr:
+            sys.stderr.write(err.message + "\n")
```

Comment:
I think this should be:
```suggestion
            sys.stderr.write("Failure loading config: " + str(err) + "\n")
```
The `err.message` property doesn't exist for all the exceptions, but "to string" seems to work for all the ones I tried.